core/invoker/src/main/scala/whisk/core/containerpool/docker/DockerContainerFactory.scala

```diff
@@ -129,3 +135,15 @@ object DockerContainerFactoryProvider extends ContainerFactoryProvider {
                                    parameters: Map[String, Set[String]]): ContainerFactory =
     new DockerContainerFactory(config, instanceId, parameters)(actorSystem, actorSystem.dispatcher, logging)
 }
+
+case class DockerContainerFactoryConfig(dataroot: Option[String]) {
+  def containerPath: File =
+    Paths
+      .get({
+        if (dataroot.isEmpty)
+          "/containers"
+        else
+          dataroot.get + "/containers"
+      })
+      .toFile
```

Comment:
Some Scala goodness: You can use `Option`s methods to transform this in a statically proven way.

```scala 
val containerPath: File = Paths.get(dataRoot.getOrElse("") + "/containers")
```

Also, I'd use a `val` since this is used quite often in production code and doesn't need to be recomputed.

Reply:
Ah yes, should have seen that sleight of hand prepending an empty string, dang.